### PR TITLE
User story 8

### LINKED ISSUE #2: User Story 8, Book Index Page can be Sorted
```
As a Visitor,
When I visit the book index page,
I should see one link each to sort the books by the following criteria:
- sorted by average rating in ascending order
- sorted by average rating in descending order
- sorted by number of pages in ascending order
- sorted by number of pages in descending order
- sorted by number of reviews in ascending order
- sorted by number of reviews in descending order
```

DO THESE STORIES FIRST:
- User Story 6, Book Index Page

### Checklist:

- [x] base code tests are written
- [x] base code is written to pass tests
- [x] edge case ("sad path") tests written
- [x] edge case code written
- [x] code is reviewed
- [x] branch is merged
- [x] all tests still pass

child of #26

child of #33

depends on #25

### PR DESCRIPTION
**Pull request by chakeresa**
This PR/branch...
 - Resolves #2 (User Story 8 - sorting the books index page)
 - Adds ability to sort books (in ascending or descending order) based on...
   - Average rating
   - Number of pages
   - Number of reviews
 - Adds links to sort the books index page in these 6 ways

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -1,6 +1,21 @@
 class BooksController < ApplicationController
   def index
-    @books = Book.all
+    case params[:sort]
```
Comment (DavisC0801) [inline app/controllers/books_controller.rb:3]: I know this isn't required but a case statement can only take one argument at a time. Not a refactor now but something to be aware of if we take multiple queries.

File: spec/models/book_spec.rb [hunk 2]
```diff
@@ -19,6 +19,49 @@
...
+    it "sorts books by number of reviews (ascending)" do
+      expect(Book.sort_by_review_count).to eq([@book_3, @book_1, @book_2])
+      expect(Book.sort_by_review_count(true)).to eq([@book_3, @book_1, @book_2])
+    end
+
+    it "sorts books by number of reviews (descending)" do
+      expect(Book.sort_by_review_count(false)).to eq([@book_2, @book_1, @book_3])
+    end
+  end
+
```
Comment (DavisC0801) [inline spec/models/book_spec.rb]: The only sad path testing I can think of is a manual query that's malformed, visit "/books_path?sort=thisisbroken. It should just return Book.all as usual but we don't have an explicit test for this.

Comment (chakeresa) [reply spec/models/book_spec.rb]: Good one -- added that test (to the feature test, since it has to do with the URL)
